pycbc/inference/models/hierarchical.py

```diff
@@ -599,3 +605,227 @@ def _loglikelihood(self):
             for stat in self.extra_stats_map[lbl]:
                 setattr(self._current_stats, stat, mstats[stat.subname])
         return logl
+
+
+class MultibandRelativeTimeDom(HierarchicalModel, DistMarg):
+    """ Hierarchical heterodyne likelihood for coherent multiband
+    parameter estimation which combines data from space-borne and
+    ground-based GW detectors coherently. Currently, this only
+    supports LISA as the space-borne GW detector.
+
+    Sub models are treated as if the same GW source (such as a GW
+    from stellar-mass BBH) is observed in different frequency band by
+    space-borne and ground-based GW detectors, then transform all
+    the parameters into the same frame in the sub model level, use
+    `HierarchicalModel` to get the joint likelihood, and marginalize
+    over all the extrinsic parameters supported by `RelativeTimeDom`
+    on the multiband likelihood level.
+
+    Note that LISA submodel only supports the `Relative` for now,
+    for ground-based detectors, please use `RelativeTimeDom`.
+    """
+    name = 'multiband_relative_time_dom'
+
+    def __init__(self, variable_params, submodels, **kwargs):
+        super().__init__(variable_params, submodels, **kwargs)
+
+        # We assume the ground-based submodel as the primary model.
+        lbl_list = list(self.submodels.keys())
+        lbl_primary = lbl_list[0]
+        self.primary_model = self.submodels[lbl_primary]
+        if self.primary_model.still_needs_det_response:
+            lbl_primary = lbl_list[1]
+            self.primary_model = self.submodels[lbl_primary]
+        self.other_models = self.submodels.copy()
+        self.other_models.pop(lbl_primary)
+        self.other_models = list(self.other_models.values())
+
+        # marginalize on multiband likelihood level
+        marginalize_phase = self.primary_model.marginalize_phase
+        variable_params, kwargs = self.setup_marginalization(
+                               variable_params,
+                               marginalize_phase=marginalize_phase,
```

Comment:
You do need to take out marginalized parameters from your variable params and from the prior. The eaisest would be to just copy from the primary model, don't recalculate.